### PR TITLE
feat: [TEO-143] add DivisionInputModal

### PR DESCRIPTION
**Pull request by OlteanuCosmin**
[TEO-143](https://wearetribus.atlassian.net/browse/TEO-143)

### CODE REVIEW

File: frontend/src/assets/locales/ro/translation.json [hunk 1]
```diff
@@ -13,5 +16,13 @@
       "dashboard": "Dashboard"
     },
     "collapse": "Restrange meniul"
+  },
+  "division": {
+    "config": {
```
Comment (birloiflorian) [inline frontend/src/assets/locales/ro/translation.json:21]: should rename this config to form.

File: frontend/src/common/utils/utils.ts [hunk 2]
```diff
@@ -2,3 +2,5 @@
 export function classNames(...classes: any[]) {
   return classes.filter(Boolean).join(' ');
 }
+
+export const NAME_REGEX = /^(?!.*[ ]{2})[\w-\săîâșțĂÎÂȘȚ]*$/;
```
Comment (birloiflorian) [inline frontend/src/common/utils/utils.ts:6]: we should do a different file in /constants called patterns.ts or regex.ts (up to you). In there we should have
export const REGEX = {
  NAME: ....
  PHONE: ....
}

File: frontend/src/components/DivisionInputModal.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+import Input from './Input';
+
+interface DivisionInputModalProps {
+  title: string;
+  onClose: () => void;
+  onSubmit: (division: DivisionInputTypes) => void;
+  defaultValue?: string;
+}
+
+type DivisionInputTypes = {
```
Comment (birloiflorian) [inline frontend/src/components/DivisionInputModal.tsx:18]: I think we should call this DivisionFormTypes, as event though it defines one input we are still using it do describe the form type.

File: frontend/src/components/DivisionInputModal.tsx [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  name: string;
+};
+
+const schema = yup
```
Comment (birloiflorian) [inline frontend/src/components/DivisionInputModal.tsx:22]: Nice one. 👍

File: frontend/src/components/DivisionInputModal.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+      <form className="space-y-6">
+        <Controller
+          key="name"
+          name="name"
+          control={control}
+          render={({ field: { onChange, value } }) => {
+            return (
+              <Input
+                type="text"
+                label="Nume"
```
Comment (birloiflorian) [inline frontend/src/components/DivisionInputModal.tsx:60]: label should have a translation.

File: frontend/src/index.css [hunk 6]
```diff
@@ -125,6 +122,13 @@
   .logo {
     @apply bg-cool-gray-700 rounded-full bg-cover;
   }
+
+  .input {
```
Comment (birloiflorian) [inline frontend/src/index.css:123]: This should not be here. We need to investigate

**Review by birloiflorian — APPROVED** [review]

File: frontend/src/common/utils/utils.ts (hunk 2 above)
Comment (OlteanuCosmin) [reply frontend/src/common/utils/utils.ts:6]: Renamed to patterns.ts

File: frontend/src/components/DivisionInputModal.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  return (
+    <Modal title={title} onClose={onClose}>
+      <form className="space-y-6">
+        <Controller
+          key="name"
+          name="name"
+          control={control}
+          render={({ field: { onChange, value } }) => {
+            return (
+              <Input
```
Comment (birloiflorian) [inline frontend/src/components/DivisionInputModal.tsx:58]: If it's part of the form and handles error we should use the FormInput component. Please revalidate.

File: frontend/src/components/DivisionInputModal.tsx [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+    control,
+    formState: { errors },
+  } = useForm<DivisionFormTypes>({
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+    resolver: yupResolver(schema),
+  });
+
+  return (
+    <Modal title={title} onClose={onClose}>
```
Comment (birloiflorian) [inline frontend/src/components/DivisionInputModal.tsx:50]: Trying to enter values in the form triggers a js error and the app fails.

**Review by birloiflorian — CHANGES_REQUESTED** [review]

**Comment by birloiflorian** [comment]
Close icon and title are not correctly aligned on mobile:
mobile alignment

**Comment by birloiflorian** [comment]
Button is on the wrong side:
Button on the wrong side

File: frontend/src/components/DivisionInputModal.tsx (hunk 8 above)
Comment (OlteanuCosmin) [reply frontend/src/components/DivisionInputModal.tsx:50]: This does not happen to me @birloiflorian

**Review by birloiflorian — APPROVED** [review]

**Pull request merged**